### ISSUE TITLE
About kp_bbox

### ISSUE THREAD

**Issue by chimeiw** [wmcnally/kapao#66]
I am reading to learn your code. I don't understand the role of kp_bbox. I see that kp_bbox is used in data augmentation, does it affect the prediction of keypoints? Could you help me ? Thanks for you work.

**Comment by wmcnally** [comment]
Where exactly are you referring to kp_bbox?

**Comment by chimeiw** [comment]
Thank you for your reply！

The “kp_bbox” in “data/hyps/hyp.kp.yaml”

The role of the rectangular box generated by kp_bbox in the train label file？
                  【kp_bbox * max(img_h, img_w) / img_w,
                  kp_bbox * max(img_h, img_w) / img_h)】

My understanding is to convert keypoint predictions into keypoint box predictions，but in yolo.py，I can't see a handle on this keypoint box.
                【if not self.training:  # inference
                if self.grid[i].shape[2:4] != x[i].shape[2:4] or self.onnx_dynamic:
                    self.grid[i] = self._make_grid(nx, ny).to(x[i].device)

                y = x[i].sigmoid()
                if self.inplace:
                    y[..., 0:2] = (y[..., 0:2] * 2. - 0.5 + self.grid[i]) * self.stride[i]  # xy
                    y[..., 2:4] = (y[..., 2:4] * 2) ** 2 * self.anchor_grid[i]  # wh

                    if hasattr(self, 'num_coords') and self.num_coords:
                        y[..., -self.num_coords:] = y[..., -self.num_coords:] * 4. - 2.
                        y[..., -self.num_coords:] *= self.anchor_grid[i].repeat((1, 1, 1, 1, self.num_coords // 2))
                        y[..., -self.num_coords:] += (self.grid[i] * self.stride[i]).repeat((1, 1, 1, 1, self.num_coords // 2))

                else:  # for YOLOv5 on AWS Inferentia https://github.com/ultralytics/yolov5/pull/2953
                    xy = (y[..., 0:2] * 2. - 0.5 + self.grid[i]) * self.stride[i]  # xy
                    wh = (y[..., 2:4] * 2) ** 2 * self.anchor_grid[i].view(1, self.na, 1, 1, 2)  # wh
                    y = torch.cat((xy, wh, y[..., 4:]), -1)
                z.append(y.view(bs, -1, self.no))
                # z.append(y)】
 Thanks for you work.

**Comment by wmcnally** [comment]
kp_bbox is the size of the keypoint bounding box labels relative to the image size. So kp_bbox = 0.05 means the keypoint bounding boxes are 5% of the input image size. This size was found to be optimal.
